app/controllers/events_controller.rb

```diff
@@ -31,4 +32,16 @@ def set_event
   def event_params
     params.require(:event).permit(:title, :content, :amount, :event_image, :date, :search, :cat)
   end
+
+  def search_events
+    @events = filter_events
+    @events = @events.where("title LIKE '%#{params[:search]}%'") if params[:search].present? && params[:search] != ''
```

Comment:
`&& params[:search] != ''`

wydaje mi się, że `params[:search].present?` już sprawdza czy to nie jest pusty String, ale mogę się mylić